projects/plugins/crm/includes/ZeroBSCRM.AdminPages.php

```diff
@@ -2162,8 +2162,44 @@ function zeroBSCRM_html_extensions() {
 		$count     = 0;
 		$idsToHide = array( 17121, 17119 );
 		if ( is_array( $extensions->paid ) ) {
+
+			$top_woo_extension_slugs = array( 'advanced-segments', 'sales-dashboard', 'automations', 'client-portal-pro', 'csv-importer-pro', 'wordpress-utilities' );
+			$extensions_to_display   = array();
+			$top_woo_extensions      = array();
+
+			// We want to prioritize the top 5 Woo modules in the list if 'woosync' is active, but otherwise alphabetize everything.
 			foreach ( $extensions->paid as $extension ) {
 
+				if ( in_array( $extension->slug, $top_woo_extension_slugs, true ) && zeroBSCRM_isExtensionInstalled( 'woo-sync' ) ) {
+					$top_woo_extensions[] = $extension;
+					usort(
+						$top_woo_extensions,
+						function (
+						$str1,
+						$str2
+						) {
+							return strcasecmp( $str1->name, $str2->name );
+						}
+					);
+				} else {
+					$extensions_to_display[] = $extension;
+					usort(
+						$extensions_to_display,
+						function (
+						$str1,
+						$str2
+						) {
+							return strcasecmp( $str1->name, $str2->name );
+						}
+					);
+				}
+			}
```

Comment:
we are ordering it on each iteration. In this case, it doesn't have a performance problem because of the size of the array. Another approach could be:

1. Before the loop, order the `$extensions->paid` array alphabetically.
2. Before the loop, create a local variable: `$has_woosync = zeroBSCRM_isExtensionInstalled( 'woo-sync' );` and the use it instead of call the function on each iteration
4. In the loop, check for `$has_woosync` first thing in the condition
5. If the extension is a top one, add it to the `$top_woo_extensions` and `continue` (without sorting it again)
6. Outside of the loop, add the extension to `$extensions_to_display` without sorting it.

Reply:
Thanks @cleacos !

> Before the loop, order the $extensions->paid array alphabetically.

I thought to myself that ordering separately would involve another loop though (iterating through each element once again), so figured it may be more performative to do it in the same loop. 

But of course, usort will take an array.... no need to do it on each iteration. Thanks!

> Before the loop, create a local variable: $has_woosync = zeroBSCRM_isExtensionInstalled( 'woo-sync' ); and the use it instead of call the function on each iteration

Oh of course, missed that - thank you!

> In the loop, check for $has_woosync first thing in the condition

This was what I was thinking originally: $has_woosync will return true more often (here I made an assumption and that, I think, is wrong, so having the array of top Woo extensions checked first means we don't need to do the additional check unnecessarily in a lot of cases (in my case that additional check was to check if WooSync was active - if the extension is not in the Woo array we just move on).

So, checking for that _first_ does make sense as it will likely return true less often. Whew. Thanks again - I will take a look at this!

Reply:
I wonder about alphabetizing the `$top_woo_extensions` array still - I tried without ordering that array and it looked unusual, with WordPress Utilities at the top. I added a new commit using your suggestions, and including sorting for the Woo array (outside of the loop) - how does that look?